lib/CourseThemeModule.js

```diff
@@ -132,64 +132,50 @@ class CourseThemeModule extends AbstractApiModule {
-        return 'font-weight: light;\n\tfont-style: italic;'
-      },
-      'woff2-normal': function () {
-        return 'font-weight: normal;\n\tfont-style: normal;'
-      },
-      'woff2-normalItalic': function () {
-        return 'font-weight: normal;\n\tfont-style: italic;'
-      },
-      'woff2-bold': function () {
-        return 'font-weight: bold;\n\tfont-style: normal;'
-      },
-      'woff2-boldItalic': function () {
-        return 'font-weight: bold;\n\tfont-style: italic;'
-      }
+    if (!fontData) {
+      return ''
     }
-    // Font files
-    if (fontData._items !== null) {
+    let fontImportCSS = ''
+    // font files
+    if (fontData._items) {
       await Promise.all(fontData._items.map(async f => {
         if (!f['font-family']) return
-
         // copy each uploaded font file
         for (const font in f._files) {
           const [data] = await assets.find({ _id: f._files[font] })
           const asset = assets.createFsWrapper(data)
           try {
-            await fs.writeFile(`${fontDest}/${f._files[font]}.${asset.data.subtype}`, await asset.read())
-
+            const relativeFontPath = `fonts/${f._files[font]}.${asset.data.subtype}`
+            const absoluteFontPath = `${fwBuild.dir}/src/theme/${fwBuild.courseData.config.data._theme}/${relativeFontPath}`
+            await fs.writeFile(absoluteFontPath, await asset.read())
             // construct font import styling
-            fontImportCSS += `@font-face {\n\tfont-family: ${f['font-family']};\n\tsrc: url('./fonts/${f._files[font]}.${asset.data.subtype}') format('${asset.data.subtype}');\n\t${fontWeightCheck[font]()}\n}\n`
+            fontImportCSS += `@font-face ${JSON.stringify({
+              'font-family': `${f['font-family']};`,
+              src: `url('./${relativeFontPath}') format('${asset.data.subtype}');`,
+              'font-weight': `${font.includes('light') ? 'light' : font.includes('bold') ? 'bold' : 'normal'};`,
+              'font-style': `${font.includes('Italic') ? 'italic' : 'normal'};`
+            }, null, 2)}`
           } catch (e) {
             this.log('error', `failed to write ${f._files[font]}, ${e.message}`)
           }
         }
       }))
-
-      fwBuild.courseData.course.data.themeVariables._font._items = undefined
+      delete fontData._items
     }
-    // Instruction style checkbox
-    if (typeof fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction['instruction-font-style'] !== null) {
-      fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction['instruction-font-style'] = fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction._isInstructionItalic ? 'italic' : 'normal'
-
-      fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction._isInstructionItalic = undefined
+    // instruction style checkbox
+    const _instruction = fontData._fontAssignments._instruction
```

Comment:
Store a local reference to fontData's _instruction attribute to allow more readable referencing